### PR TITLE
Replace _.each and for-in with for loop in hot paths

### PR DESCRIPTION
**Pull request by gaearon**
This is my first PR extracted from experiments in #428.
This alone gives me 24% time improvement (from 212ms on my benchmark to 160ms).

### CODE REVIEW

**Comment by gaearon** [comment]
With second commit, 29% improvement over master (150ms instead of 212ms).

**Comment by chicoxyzzy** [comment]
:+1: looks good

File: lib/6to5/generation/position.js [hunk 1]
```diff
@@ -1,33 +1,27 @@
...
 }
 
 Position.prototype.push = function (str) {
-  var self = this;
-
-  _.each(str, function (cha) {
-    if (cha === "\n") {
-      self.line++;
-      self.column = 0;
+  for (var i = 0; i < str.length; i++) {
```
Comment (eventualbuddha) [inline lib/6to5/generation/position.js]: I'm not sure if it makes a difference, but does caching the length here help? i.e. `for (var i = 0, length = str.length; i < length; i++) {`

File: lib/6to5/generation/position.js [hunk 2]
```diff
@@ -1,33 +1,27 @@
...
+  }
 };
 
 Position.prototype.unshift = function (str) {
-  var self = this;
-
-  _.each(str, function (cha) {
-    if (cha === "\n") {
-      self.line--;
+  for (var i = 0; i < str.length; i++) {
```
Comment (eventualbuddha) [inline lib/6to5/generation/position.js]: Ditto.

File: lib/6to5/generation/position.js (hunk 1 above)
Comment (gaearon) [inline lib/6to5/generation/position.js]: No, it doesn't appear so. I ran it and it's the same speed.
I remember reading V8 does that optimization by itself.

Comment (eventualbuddha) [inline lib/6to5/generation/position.js]: Nevermind, I just saw the discussion on #428.

Comment (sairion) [inline lib/6to5/generation/position.js]: AFAIK caching built-in length won't make big difference in modern JS engine like V8. See: http://mrale.ph/blog/2014/12/24/array-length-caching.html

**Comment by eventualbuddha** [comment]
:+1:

**Comment by gaearon** [comment]
I added a few minor impovements: https://github.com/gaearon/6to5/commit/5783973734d3b03c0f3809e16e0b51af401a4636

In my testing, they shave off about 5-10ms from my benchmark.

I'm afraid I can't find other low-hanging optimization fruit other than this and #433 so I'll be glad if people smarter than me continued with this. :-)

**Comment by thejameskyle** [comment]
:+1:

**Comment by sebmck** [comment]
Awesome, thanks for this!

**Pull request merged**